### PR TITLE
feat: git integration

### LINKED ISSUE #25: Git branch string
Could a string token be added for showing the current git branch?

### PR DESCRIPTION
**Pull request by Tim-Tech-Dev**
- Add git.d.ts from microsoft/vscode
- Add gitManager.ts where we extract the required strings
- Dispose of gitManager.ts in extension.ts
- Add {{git_repo}} and {{git_branch}} placehoders to constants.ts and activity.ts
- Remove hacky getGitRepo function from utils.ts
- Update package.json config descriptions
- Update READEME.md and fix some typos

fix #25

### CODE REVIEW

**Comment by Tim-Tech-Dev** [comment]
Currently, very unstable git repo detection. Will need some work.

**Comment by LeonardSSH** [comment]
👍

File: README.md [hunk 1]
```diff
@@ -52,6 +52,8 @@ Remember to 🌟 this GitHub if you 💖 it.
 
 > All texts are fully customizable using variables and a multitude of config options
 
+<!--You know other people can read this too?-->
```
Comment (LeonardSSH) [inline README.md:55]: easter egg 🤣🤣

**Comment by Tim-Tech-Dev** [comment]
I would say you can check this PR now. The git feature is working reliable now. But I would like to make a second version if I have the time to do so. Because we could add/remove listeners and update the RPC only if the configuration needs it.

[onConfigurationChange](https://code.visualstudio.com/api/references/vscode-api#workspace.onDidChangeConfiguration)=>
> choose from:
> listeners for files
> listeners for git
> listeners for workspaces

To reduce the amount of updates caused by file changes or git if there is only a need for the current workspace or just a custom string.

**Comment by Tim-Tech-Dev** [comment]
Just found a bug in the code. It remains permanently at idle.

**Comment by Tim-Tech-Dev** [comment]
Thanks for the ping! I may have forgotten this PR… In any case, I tried to reproduce the bug I mentioned, but I can't reproduce it. I can confirm, however: The feature works. You may merge it now. If any issue crops up, I'll be in touch.

**Comment by LeonardSSH** [comment]
I'm glad to hear that, I'll give this PR a pull to test if everything is ok before merge it.

**Pull request merged**